sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcUtil.java

```diff
@@ -0,0 +1,179 @@
+                  (element, ps, i, fieldWithIndex) ->
+                      ps.setBigDecimal(i + 1, element.getDecimal(fieldWithIndex.getIndex())))
+              .put(
+                  Schema.TypeName.FLOAT,
+                  (element, ps, i, fieldWithIndex) ->
+                      ps.setFloat(i + 1, element.getFloat(fieldWithIndex.getIndex())))
+              .put(
+                  Schema.TypeName.DOUBLE,
+                  (element, ps, i, fieldWithIndex) ->
+                      ps.setDouble(i + 1, element.getDouble(fieldWithIndex.getIndex())))
+              .put(
+                  Schema.TypeName.DATETIME,
+                  (element, ps, i, fieldWithIndex) ->
+                      ps.setTimestamp(
+                          i + 1,
+                          new Timestamp(
+                              element.getDateTime(fieldWithIndex.getIndex()).getMillis())))
+              .put(
+                  Schema.TypeName.BOOLEAN,
+                  (element, ps, i, fieldWithIndex) ->
+                      ps.setBoolean(i + 1, element.getBoolean(fieldWithIndex.getIndex())))
+              .put(Schema.TypeName.BYTES, createBytesCaller())
+              .put(
+                  Schema.TypeName.INT32,
+                  (element, ps, i, fieldWithIndex) ->
+                      ps.setInt(i + 1, element.getInt32(fieldWithIndex.getIndex())))
+              .put(Schema.TypeName.STRING, createStringCaller())
+              .build());
+
+  /** PreparedStatementSetCaller for Schema Field Logical types. * */
+  public static JdbcIO.PreparedStatementSetCaller getPreparedStatementSetCaller(
+      Schema.FieldType fieldType) {
+    switch (fieldType.getTypeName()) {
+      case ARRAY:
+        return (element, ps, i, fieldWithIndex) -> {
+          ps.setArray(
+              i + 1,
+              ps.getConnection()
+                  .createArrayOf(
+                      fieldType.getCollectionElementType().getTypeName().name(),
+                      element.getArray(fieldWithIndex.getIndex()).toArray()));
+        };
+      case LOGICAL_TYPE:
+        {
+          String logicalTypeName = fieldType.getLogicalType().getIdentifier();
+          JDBCType jdbcType = JDBCType.valueOf(logicalTypeName);
+          switch (jdbcType) {
+            case DATE:
+              return (element, ps, i, fieldWithIndex) -> {
+                ps.setDate(
+                    i + 1, new Date(element.getDateTime(fieldWithIndex.getIndex()).getMillis()));
+              };
+            case TIME:
+              return (element, ps, i, fieldWithIndex) -> {
+                ps.setTime(
+                    i + 1, new Time(element.getDateTime(fieldWithIndex.getIndex()).getMillis()));
+              };
+            case TIMESTAMP_WITH_TIMEZONE:
+              return (element, ps, i, fieldWithIndex) -> {
+                ps.setTimestamp(
+                    i + 1,
+                    new Timestamp(element.getDateTime(fieldWithIndex.getIndex()).getMillis()));
```

Comment:
As this is a `timezoned` type, is time zone handled here?

Reply:
updated - UTC timezone is set as default when saving timestamp with timezone.

Reply:
I am thinking `element.getDateTime` will return a Joda Datetime, which has timezone information already?

Reply:
timestamp is the moment in time I don't think timezone is needed here because when user is going to read data from DB (timestamp field) then it needs to be converted at that point.
Please check this link https://stackoverflow.com/questions/15994450/joda-datetime-to-timestamp-conversion/15994657#15994657
can you share something related to your point? may be I am missing something here

Reply:
There are two TIMESTAMP: `TIMESTAMP without TIMEZONE`(sometimes I saw it's called DATETIME), `TIMESTAMP with TIMEZONE`.

In this implementation, it's processing a `TIMESTAMP with TIMEZONE` type, which definitely has a timezone information that matters. I think here are two concepts matter (which is also covered by your link): an absolute point in time and an offset. Offset (specified by timezone) should be lost for `TIMESTAMP with TIMEZONE`


I find this SO question is useful: https://stackoverflow.com/questions/14070572/is-java-sql-timestamp-timezone-specific

Reply:
used setTimestamp with calendar instance for timezone
done